src/forms/TextInput.jsx

```diff
@@ -123,42 +81,32 @@ export const TextInput = (props: Props): React$Element<*> => {
 	}
 
 	return (
-		<div className="inputContainer">
+		<div>
 			{label && (
-				<label
-					className={classNames.label}
-					htmlFor={id}
-					data-requiredtext={required && requiredText}
-				>
+				<label htmlFor={id || name}>
 					{label}
+					{required ? <span> {requiredText}</span> : ''}
```

Comment:
Could be shortened to `{required && (<span>{requiredText}</span>)}

Reply:
It needs that extra space, but yes, I agree `&&` probably better.